ocfweb/docs/docs/services/hpc.md

```diff
@@ -1,65 +1,115 @@
 * `squeue`: Used to view running and queued jobs.
 
-### Dependencies
-For managing application dependencies, one currently has two options:
+## Dependencies
+For managing application dependencies, you currently have two options:
 
-First you could use a virtual environment if you are using Python packages. To create a virtual environment navigate to your home directory and run the following commands
+### Virtual Environments
 
-    virtualenv -p python3.5 venv
-    . venv/bin/activate
+First you could use a [virtual environment][venv] if you are using Python packages. To create a virtual environment navigate to your home directory and run the following commands:
+
+```
+virtualenv --system-site-packages -p python3 venv
+. venv/bin/activate
+```
 
 This will allow you to *pip install* any Python packages that the OCF does not already have for your program.
 
-For those who need access to non-python dependencies or have already integrated their program into Docker, the second option is to use Singularly containers. [Singularity][singularity] is a containerization platform developed at Laurence Berkeley National Laboratory that is designed specifically for HPC environments. To read more about the benefits of Singularity you can look [Here][singularity_article]. Using a container is quite simple, below are a few commands to help getting started.
+### Singularity
+
+For those who need access to non-Python dependencies or have already integrated their program into Docker, the second option is to use Singularity containers. [Singularity][singularity] is a containerization platform developed at Lawrence Berkeley National Laboratory that is designed specifically for HPC environments. To read more about the benefits of Singularity you can look [here][singularity_article]. We suggest a particular workflow, which will help simplify deploying your program on our infrastructure.
+
+#### Installing
+
+If you are running linux on your system, you can install Singularity from the official apt repos:
+
+```
+sudo apt install singularity-container
+```
+
+If you are running Mac you can look [here][mac_install], or Windows [here][win_install].
+
+#### Building Your Container
+
+```
+singularity build --sandbox ./my_container docker://ubuntu
+```
+This will create a Singularity container named `my_container`. If you are still testing out your container on your system, we suggest you use the `--sandbox` option. This will allow you to install new packages in your container as you need them without needing sudo privileges. The `docker://ubuntu` option notifies Singularity to bootstrap the container from the official Ubuntu docker container on [Docker Hub][docker_hub]. There is also a [Singularity Hub][singularity_hub], from which you can directly pull Singularity images in a similar fashon. We also have some pre-built containers that you may use to avoid having to build your own. They are currently located at `/home/containers`.
```

Comment:
> This will allow you to install new packages in your container as you need them without needing sudo privileges.

Sounds ambiguous to me: sudo privileges inside or outside the container?

Reply:
+1. Just put a stop after "as you need them."

Reply:
may also want to clarify where exactly '/home/containers' is located, i.e. on segfault